pages/wiki/openembedded.md

```diff
@@ -3,10 +3,245 @@ title: OpenEmbedded
-<p>A recipe describe how to fetch, patch, configure, compile, install, package (in .rpm, .deb or .ipk) and test a piece of software. Bitbake handles all those operations and dependencies between them and between recipes. In the end it can generate images or SDK for multiple targets.</p>
-<p>Contributing to the AsteroidOS’s OE architecture mostly consists in maintaining the <a href="https://github.com/AsteroidOS/meta-asteroid">meta-asteroid</a> and meta-*-hybris repositories which contain the recipes related to Asteroid and the watches BSPs.</p>
-<p>In depth info can be found in <a href="https://docs.yoctoproject.org/index.html">the Yocto Project's manual</a>.</p>
+# OpenEmbedded
+
+[OpenEmbedded](http://www.openembedded.org/wiki/Main_Page) (OE) is the build system used by AsteroidOS. It provides a way to easily maintain embedded Linux distributions. OE is also used by project such as WebOS-Ports, SHR or Ångström. OpenEmbedded is supported by the [Yocto project](https://www.yoctoproject.org/) from the Linux Foundation and provides a rock-solid basis for AsteroidOS.
+
+OpenEmbedded uses the bitbake tool which is developed in python to generate images (like asteroid-image) from packages (like asteroid-launcher-dev) built from recipes (ex: asteroid-launcher_git.bb) provided in layers (ex: meta-asteroid).
+
+Layers are git repositories containing a bunch of related recipes, example of layers include board support packages (BSPs) like [meta-dory-hybris](https://github.com/AsteroidOS/meta-smartwatch/tree/master/meta-dory) or meta-rockchip which describe a way to support a new machine, but also UI layers like meta-xfce or meta-gnome which describe the building process of graphic components.
+
+A recipe describes how to fetch, patch, configure, compile, install, package (in .rpm, .deb or .ipk) and test a piece of software. Bitbake handles all those operations and dependencies between them and between recipes. In the end it can generate images or SDK for multiple targets.
+
+Contributing to the AsteroidOS’s OE architecture mostly consists in maintaining the [meta-asteroid](https://github.com/AsteroidOS/meta-asteroid) and [meta-smartwatch](https://github.com/AsteroidOS/meta-smartwatch) repositories which contain the recipes related to Asteroid and the watches' BSPs.
+In depth info can be found in the [Yocto Project's manual](https://docs.yoctoproject.org/index.html)
+
 <img src="{{assets}}/img/openembedded.png" style="width:100%"/>
+
+
+# A Quick Tutorial
+
+---
+
+
+The rest of this description assumes that you have already read and followed the [Building AsteroidOS]({{rel 'wiki/building-asteroidos'}}) instructions, and that you have built AsteroidOS for at least one platform (a particular watch or the emulator).
+
+This guide is for those who wish to go deeper and would like to alter or augment the AsteroidOS software.
```

Comment:
I had them in that order originally, but "Sharing your changes" applies equally to both "Changing an Existing Application" and "Adding a New Application" so it seemed to make more sense to me to put it at the end.

Reply:
> > Another excellent PR.
> > In this one I would also like to add that there are a few `double spaces` here, instead of mentioning them individually.
> 
> Double spaces after a period are an artifact of the way I learned to type fifty years ago and not a habit I'm actively trying to alter, although I understand the conventions have (relatively) recently changed. However, for this it truly doesn't matter since both single and double spaces are rendered _exactly_ the same way on the web site.

I've changed to all single spaces per convention.  My fingers still type two spaces, but I search and replace pretty well.

Reply:
I totally agree on the current structure.
This was more about adding a short paragraph that introduces the next sections. As a way to guide the reader.
It could be a simple list as well:

This tutorial will introduce you to the following topics:
- *Changing an existing application*: You don't like how something behaves or want to fix an issue in an existing application? This will introduce you to the tooling used to allow you to make these changes.
- *Adding a new application*: Use the tooling to create a new application recipe and test it on a watch.
- *Sharing your changes*: You've made changes on your local build of AsteroidOS. How do you share these changes? This will be explained in this part of the tutorial.

Reply:
I like this idea and have added your text to the page, verbatim.  Thanks!